### ISSUE TITLE
World.CurrentDate does not return

### ISSUE THREAD

**Issue by RainerHilmer** [crosire/scripthookvdotnet#109]
I tried to receive the current day of the week but this little spike demonstrates World.CurrentDate does not return because the spike can't make it to UI.ShowSubtitle anymore.

using System.Windows.Forms;
using Cyron43.GtaV.Common;
using GTA;

namespace GTAVSpikes
{
   /// <summary>
   /// Do whatever you currently need to research here.
   /// </summary>
   public class DynamicSpike : Script
   {
      public DynamicSpike()
      {
         KeyDown += OnKeyDown;
         UI.Notify("Spike is running.");
      }

      private void OnKeyDown(object sender, KeyEventArgs e)
      {
         var testKey = new KeyContainer
                       {
                          Alt = false,
                          Ctrl = false,
                          Key = Keys.Space,
                          Name = "Test",
                          Shift = false
                       };
         if(KeyHandlingCommons.KeyIs(testKey, e))
            DoTest();
      }

      private void DoTest()
      {
         UI.Notify("DoTest called.");
         var weekDay = World.CurrentDate.DayOfWeek;
         UI.ShowSubtitle("Today is " /* + weekDay*/); //Shows only when the line before is commented out.
      }
   }
}

**Comment by JohnnyCrazy** [comment]
Any errors in the logs?

**Comment by IncoCode** [comment]
Because gta date != "normal" date. In gta all months have 31 day. If your current game month is February and current game date is 30 - DateTime can't parse that.
We should use our own class for these purposes (something like [this](https://github.com/IncoCode/AccountInBank-GTA5/blob/master/Code/AccountInBank/Model/GTADate.cs)).

**Comment by RainerHilmer** [comment]
Log:
[17:18:41] [ERROR] Caught unhandled exception:
System.ArgumentOutOfRangeException: Die Parameter "Year", "Month", und "Day" beschreiben eine nicht darstellbare DateTime.
   bei System.DateTime.DateToTicks(Int32 year, Int32 month, Int32 day)
   bei GTA.World.get_CurrentDate()
   bei GTAVSpikes.DynamicSpike.DoTest()
   bei GTA.Script.MainLoop()

**Comment by IncoCode** [comment]
Also, month can be from 0 to 11.

**Comment by IncoCode** [comment]
One more: after 31.11.2013 new year becomes... 2010.

**Comment by RainerHilmer** [comment]
Doesn't crash anymore but it returns the wrong weekday.

**Comment by IncoCode** [comment]
@RainerHilmer because your current gta date corresponds to rules of normal date.
We donэt need to use DateTime object.

**Comment by JohnnyCrazy** [comment]
Fixed in `feature/date`. Will be merged once 2.0 is ready

**Comment by RainerHilmer** [comment]
:+1:

**Comment by crosire** [comment]
Yeah, they should only be closed once merged into the development branch.
